packages/@aws-cdk/aws-ec2/lib/vpc-endpoint.ts

```diff
@@ -0,0 +1,425 @@
+import iam = require('@aws-cdk/aws-iam');
+import cdk = require('@aws-cdk/cdk');
+import { Connections, IConnectable } from './connections';
+import { CfnVPCEndpoint } from './ec2.generated';
+import { SecurityGroup } from './security-group';
+import { VpcSubnet } from './vpc';
+import { IVpcNetwork, IVpcSubnet, SubnetSelection } from './vpc-ref';
+
+/**
+ * A VPC endpoint.
+ */
+export interface IVpcEndpoint extends cdk.IConstruct {
+  /**
+   * The VPC endpoint identifier.
+   */
+  readonly vpcEndpointId: string;
+}
+
+/**
+ * A VPC gateway endpoint.
+ */
+export interface IVpcGatewayEndpoint extends IVpcEndpoint {
+  /**
+   * Exports this VPC endpoint from the stack.
+   */
+  export(): VpcGatewayEndpointImportProps;
+}
+
+/**
+ * The type of VPC endpoint.
+ */
+export enum VpcEndpointType {
+  /**
+   * Interface
+   */
+  Interface = 'Interface',
+
+  /**
+   * Gateway
+   */
+  Gateway = 'Gateway'
+}
+
+/**
+ * A VPC endpoint service.
+ */
+export class VpcEndpointService {
+  constructor(public readonly name: string, public readonly type: VpcEndpointType) {}
+}
+
+/**
+ * A VPC endpoint AWS service.
+ */
+export class VpcEndpointAwsService extends VpcEndpointService {
+  public static readonly SageMakeNotebook = new VpcEndpointAwsService('sagemaker', VpcEndpointType.Interface, 'aws.sagemaker');
+  public static readonly CloudFormation = new VpcEndpointAwsService('cloudformation');
+  public static readonly CloudTrail = new VpcEndpointAwsService('cloudtrail');
+  public static readonly CodeBuild = new VpcEndpointAwsService('codebuild');
+  public static readonly CodeBuildFips = new VpcEndpointAwsService('codebuil-fips');
+  public static readonly CodeCommit = new VpcEndpointAwsService('codecommit');
+  public static readonly CodeCommitFips = new VpcEndpointAwsService('codecommit-fips');
+  public static readonly CodePipeline = new VpcEndpointAwsService('codepipeline');
+  public static readonly Config = new VpcEndpointAwsService('config');
+  public static readonly DynamoDb = new VpcEndpointAwsService('dynamodb', VpcEndpointType.Gateway);
+  public static readonly Ec2 = new VpcEndpointAwsService('ec2');
```

Comment:
Somehow I would rather name those `EC2`, `ECR`, `SNS`, ... The PascalCased name hurts my eyes.

Reply:
Agree with you but when looking at other packages I find lots of PascalCased names:
https://github.com/awslabs/aws-cdk/blob/7001f7723cde611f0d8cd9c182f92120b09579c3/packages/%40aws-cdk/aws-sns/lib/subscription.ts#L89-L93
https://github.com/awslabs/aws-cdk/blob/7001f7723cde611f0d8cd9c182f92120b09579c3/packages/%40aws-cdk/aws-applicationautoscaling/lib/scalable-target.ts#L227-L246

Reply:
Your call

Reply:
Hmm. I guess consistency is good. 🤷🏻‍♂️
Thanks for pointing those out 👍🏻

Reply:
Pascal case is our standard. Otherwise jsii languages cannot delineate words when they convert cases.